Modules/mmapmodule.c

```diff
@@ -1117,6 +1117,46 @@ mmap_mmap_seek_impl(mmap_object *self, Py_ssize_t dist, int how)
     return NULL;
 }
 
+/*clinic*/
+
+/*[clinic input]
+mmap.mmap.set_name
+
+    name: str
+    /
+
+[clinic start generated code]*/
+
+static PyObject *
+mmap_mmap_set_name_impl(mmap_object *self, const char *name)
+/*[clinic end generated code: output=1edaf4fd51277760 input=6c7dd91cad205f07]*/
+{
+#if defined(MAP_ANONYMOUS) && defined(__linux__)
+    const char *prefix = "cpython:mmap:";
```

Comment:
Is it really needed to include `mmap:` in the name? Why not just:

```suggestion
    const char *prefix = "cpython:";
```

Reply:
An alternative is to even omit the prefix, and just the name to *name*.

Reply:
> Is it really needed to include mmap: in the name? Why not just:

I think "yes", If you don't have strong preference, I would like to keep it as is, because it's called from mmap module.
Petr suggested a similar convention. https://github.com/python/cpython/pull/142079#discussion_r2576513426

It would be easy to filter everything with "cpython:" prefix.

Reply:
Also able to avoid unintended conflict between already used names.